### PR TITLE
Fix #13581: Opening the Options menu kills performance

### LINKED ISSUE #13581: Opening the Options Menu kills performance
**OS:** Arch Linux
**Version:** 0.3.2-198
**Commit/Build:** 9ef493198

Opening the Options Menu, regardless of Rendering option, cause frame rates to drop to nearly 1/10th of expected performance.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Observe FPS
2. Open Options Menu
3. Observe again

### PR DESCRIPTION
**Pull request by Gymnasiast**

### CODE REVIEW

File: src/openrct2-ui/UiContext.Linux.cpp [hunk 1]
```diff
@@ -251,10 +253,15 @@ namespace OpenRCT2::Ui
             return result;
         }
 
-        bool HasFilePicker() const override
+        bool HasFilePicker() override
```
Comment (duncanspumpkin) [inline src/openrct2-ui/UiContext.Linux.cpp:256]: Think you would normally mark the cache as mutable rather than change the const of the function

Comment (Gymnasiast) [reply src/openrct2-ui/UiContext.Linux.cpp:256]: Sounds better, how can I do that?

Comment (duncanspumpkin) [reply src/openrct2-ui/UiContext.Linux.cpp:256]: https://en.cppreference.com/w/cpp/language/cv 
Just add mutable before the `std::optional<bool> _hasFilePicker = std::nullopt;`

File: src/openrct2-ui/windows/Options.cpp [hunk 2]
```diff
@@ -2280,6 +2281,7 @@ rct_window* window_options_open()
     w->page = WINDOW_OPTIONS_PAGE_DISPLAY;
     w->frame_no = 0;
     WindowInitScrollWidgets(w);
+    _hasFilePicker = OpenRCT2::GetContext()->GetUiContext()->HasFilePicker();
```
Comment (IntelOrca) [inline src/openrct2-ui/windows/Options.cpp:2284]: You can probably revert the changes to this file now as you cache in the context instead.

Comment (Gymnasiast) [reply src/openrct2-ui/windows/Options.cpp:2284]: Wouldn't that increase the overhead again?

Comment (duncanspumpkin) [reply src/openrct2-ui/windows/Options.cpp:2284]: The overhead is in the call that you have now cached. So this is going to be pretty cheap to call

File: src/openrct2-ui/UiContext.Linux.cpp [hunk 3]
```diff
@@ -35,6 +35,8 @@ namespace OpenRCT2::Ui
     class LinuxContext final : public IPlatformUiContext
     {
     private:
+        mutable std::optional<bool> _hasFilePicker = std::nullopt;
```
Comment (AaronVanGeffen) [inline src/openrct2-ui/UiContext.Linux.cpp:38]: If no other methods are using this, you could make this a static variable in the HasFilePicker method instead.

Comment (IntelOrca) [reply src/openrct2-ui/UiContext.Linux.cpp:38]: That would break a few principals.

Comment (AaronVanGeffen) [reply src/openrct2-ui/UiContext.Linux.cpp:38]: How so? What am I missing?

Comment (IntelOrca) [reply src/openrct2-ui/UiContext.Linux.cpp:38]: Lets say you are writing a test for this class, you would get state shared between one test and the next.

Comment (AaronVanGeffen) [reply src/openrct2-ui/UiContext.Linux.cpp:38]: Right, and that wouldn't happen for properties. Gotcha. Thanks.
